### ISSUE TITLE
Get full gradients in Deepspeed 0.6.0

### ISSUE THREAD

**Issue by leenachennuru** [microsoft/DeepSpeed#4917]
Hi, 

Is there a way to obtain the full gradient in Deepspeed 0.6.0? This version does not have the safe_full_get_grad function which was introduced in later versions but I was wondering if there is a some other way to obtain the full gradients in DS 0.6.0. 

Any information on this would be appreciated!

Thanks.

**Comment by loadams** [comment]
Hi @leenachennuru - is there a reason that you are restricted to using DeepSpeed 0.6.0 as it was released almost 2 years ago?

**Comment by leenachennuru** [comment]
Yes, I built my pipeline around it and it would unfortunately be a rather large overhead to switch to a later version.

**Comment by ShadenSmith** [comment]
Hi @leenachennuru, thanks for using DeepSpeed! We are always striving to improve and regularly ship bugfixes and performance improvements. I would recommend trying to use a later version of DeepSpeed in order to get those improvements too :-).

As for your current issue, the `safe_get_full_grad` utility looks pretty short, perhaps it could be ported to your pipeline to try? We've released plenty of system improvements that include data structure changes and refactoring though, so no guarantees that it'll work out of the box. I think updating to a current DeepSpeed release would be the safer and better option.
